### PR TITLE
Cats and pugs are 100% more adorable (includes: refactoring, map stuff and minor collar bugfix)

### PR DESCRIPTION
**Pull request by harmonyn**
The tyranny of corgis being the only pets that respond like actual pets is ogre. List of changes:
- Pet collars now give pets proper names. No more "The testing yaps happily!" 
- Runtime, and all cats, will now purr when you pet her and hiss when you kick her. You monster.
- Pugs will also respond to pets and attacks, the same way as corgis do.
- SOMEWHAT IMPORTANT: To accomplish the above, and to remove some duplicated code, corgis and pugs now share the same parent type dog:
- corgi.dm is now dog.dm
- pug.dm no longer exists, everything there (which isn't much) is merged into dog.dm
- The path for corgis is now /mob/living/simple_animal/pet/dog/corgi; the path for pugs is now /mob/living/simple_animal/pet/dog/pug; the path for corgi puppies is now /mob/living/simple_animal/pet/dog/corgi/puppy.

This repathing affects other areas of the code where corgis and/or Ian appears, notably:
- Critter crates
- Experimentor
- Facehugging
- Holidays
- Admin topic.dm (also, admins can now become pugs, they couldn't before)
- QUITE IMPORTANT: All included maps, because Ian spawns on them. 

**YO MAPPERS UPDATE YOUR MAPS TO THE NEW PATH OR ELSE IAN WILL NOT WORK** (also it probably won't work in general)

merge conflict is because of the map edit, the map change in here is literally just Ian's one line so future conflicts should be easy to resolve

### CODE REVIEW

File: tgstation.dme [hunk 1]
```diff
@@ -6,6 +6,66 @@
 
 // BEGIN_FILE_DIR
 #define FILE_DIR .
+#define FILE_DIR "code"
```
Comment (CobaltAnt) [inline tgstation.dme]: u didn't do the thing 

https://tgstation13.org/phpBB/viewtopic.php?f=5&t=321&p=106881

Comment (Jordie0608) [inline tgstation.dme]: To get rid of these, in DM go to buuld > preferences and untick 'Automatically set FILE_DIR for subdirectories' then recommit the dme.

File: code/modules/mob/living/simple_animal/friendly/pug.dm [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
-	..()
-
-	if(!stat && !resting && !buckled)
-		if(prob(1))
-			emote("me", 1, pick("chases its tail."))
-			spawn(0)
-				for(var/i in list(1,2,4,8,4,2,1,2,4,8,4,2,1,2,4,8,4,2))
-					dir = i
-					sleep(1)
+/mob/living/simple_animal/pet/pug
```
Comment (Jordie0608) [inline code/modules/mob/living/simple_animal/friendly/pug.dm]: This file isn't actually gone.

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 3]
```diff
@@ -1,542 +1,568 @@
-//Corgi
```
Comment (Jordie0608) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: This looks like a line ending issue thats causing git to report every line as changed. The file should be CRLF, so you'll need to convert it back to that from likely LF that it got automagically changed it, or vice versa if my memory is wrong. You can do it with Notepad++.

**Comment by bgobandit** [comment]
all jordie's stuff should be resolved now. merge conflict almost certainly due to the umpteen map edits in PRs right now

**Comment by Niknakflak** [comment]
You could just do the map edit like everyone else does when they change stuff like this that way there isn't that gap time where shits broken before mappers update.

Also, I hate you forever for getting rid of corgi.dm
Pugs don't deserve to act real

**Comment by bgobandit** [comment]
it's fixed

IT'S CATTENING

**Pull request closed without merging**